app/src/main/java/org/mozilla/rocket/download/DownloadInfoViewModel.kt

```diff
@@ -19,11 +21,26 @@ class DownloadInfoViewModel(private val repository: DownloadInfoRepository) : Vi
     var isOpening = false
 
     private val downloadInfoBundle: DownloadInfoBundle = DownloadInfoBundle(ArrayList(), -1, -1)
-    private var handler: Handler? = null
+
     private var itemCount = 0
     private var isLoading = false
     private var isLastPage = false
 
+    private val handlerThread: HandlerThread = HandlerThread("download-progress")
```

Comment:
move the HandlerThread to repository

Reply:
After discussion, we will leave HandlerThread in ViewModel